client/server/render/index.js

```diff
@@ -134,10 +134,11 @@ export function render( element, key = JSON.stringify( element ), req ) {
 						extra: {
 							message: ex.message,
 							stack: ex.stack,
+							userId: req.context?.user?.ID,
```

Comment:
The `/logstash` endpoint supports a `user_id` field:
```js
logToLogstash( {
  feature: 'calypso_ssr',
  message: ...,
  user_id: ...
} );
```
The difference is that the `user_id` will be logged as standard, indexed and searchable field, instead of being buried in the `extra` field.

Also, we should include it only if it's known. Or maybe the JSON serialization automatically excludes fields with `undefined` or `null` value?

Optional chaining nit: `req.context` is always present and doesn't need to be guarded by `?.`. Only the `user` field is optional and accessing `user.ID` may fail:
```js
req.context.user?.ID
```

Reply:
Good point about using `user_id`.

As far as I can tell, the user id is _never_ known at this point because this function is only run when there is no logged in user. But in any case, the serialization excludes `undefined` fields

Reply:
> As far as I can tell, the user id is never known at this point because this function is only run when there is no logged in user.

That's true, SSR is active only for logged out pages. That can change in future though 🙂